lms/djangoapps/learner_home/recommendations/test_views.py

```diff
@@ -155,21 +143,29 @@ def test_amplitude_api_unexpected_error(self):
 
     @override_waffle_flag(ENABLE_LEARNER_HOME_AMPLITUDE_RECOMMENDATIONS, active=True)
     @mock.patch(
-        "lms.djangoapps.learner_home.recommendations.views.get_personalized_course_recommendations"
+        "lms.djangoapps.learner_home.recommendations.views.get_amplitude_course_recommendations"
     )
-    @mock.patch("lms.djangoapps.learner_home.recommendations.views.get_course_data")
+    @mock.patch("lms.djangoapps.learner_home.recommendations.views.filter_recommended_courses")
     def test_get_course_recommendations(
-        self, mocked_get_course_data, mocked_get_personalized_course_recommendations
+        self, filter_recommended_courses_mock, get_amplitude_course_recommendations_mock
     ):
         """
         Verify API returns course recommendations.
         """
-        mocked_get_personalized_course_recommendations.return_value = [
+        get_amplitude_course_recommendations_mock.return_value = [
             False,
             True,
             self.recommended_courses,
         ]
-        mocked_get_course_data.return_value = self.course_data
+        filtered_course = []
```

Comment:
Can we write a private function for course filteration to reduce redundancy 
```suggestion
        def _get_filtered_courses():
            filtered_course = []
            for course_key in self.recommended_courses[:5]:
              filtered_course.append({
                  "course_key": course_key,
                  "title": f"Title for {course_key}",
                  "logo_image_url": "https://www.logo_image_url.com",
                  "marketing_url": "https://www.marketing_url.com",
                })
            return filtered_course
```